### ISSUE TITLE
Add detail to The Scheduler Run Sequence

### ISSUE THREAD

**Issue by tom131313** [wpilibsuite/frc-docs#2297]
[Regarding Document](https://github.com/wpilibsuite/frc-docs/blob/main/source/docs/software/commandbased/command-scheduler.rst)

An important use, of a subsystem as a Finite State Machine and its dependent commands as States of that FSM upon a trigger/button event to initiate a transition from one state to another, depends on the following sequence:

_the `end` method of the current (interrupted) command (state) is run before the `initialize` method of the new (interrupting) command (state)._

Using state exit and enter like this somewhat special case depends on the user correctly specifying a FSM as a subsystem and that subsystem being a requirement of all the commands that reflect states of that FSM and the commands can be interrupted by subsequent commands.

WPILib supports that sequence and adding detail to the documentation to describe that case works would help users.

[ChiefDelphi reference](https://www.chiefdelphi.com/t/commandbase-subsystembase-depreciation/438975/166)

**Comment by Starlight220** [comment]
Just to make sure I understand, the request here is to clarify that interrupted command `end` is called before interrupting command `init`?
Anything more than that?

**Comment by tom131313** [comment]
Yes and all this happens in the **same timestep** to provide a bump-less transition to a new state:
1. event triggers new command with dependency on a subsystem
2. `end()` the running command for that subsystem
3. `initialize()` new command for that subsystem

In ChiefDelphi I made this comment about an example application of this feature is to make a smooth, safe motor speed transition (assuming the motor `set()` speed is also properly handled and that might not be obvious coding).
